test_report/src/report.rs

```diff
@@ -0,0 +1,247 @@
+use chrono::prelude::*;
+#[cfg(feature = "ruby")]
+use magnus::{value::ReprValue, Module, Object};
+use prost_wkt_types::Timestamp;
+use proto::test_context::test_run::{TestCaseRun, TestCaseRunStatus, TestResult, UploaderMetadata};
+use std::cell::RefCell;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct TestReport {
+    test_result: TestResult,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+#[cfg_attr(feature = "ruby", magnus::wrap(class = "TestExecution"))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TestExecution {
```

Comment:
This is 1:1 with TestCaseRun in proto, right? Any reason we can't unify/codegen this

Reply:
It isn't 1:1. The bindings also make that tricky.